devicetypes/smartthings/philio-multiple-sound-siren.src/philio-multiple-sound-siren.groovy

```diff
@@ -0,0 +1,314 @@
+					descriptionText: "This device failed to complete the network security key exchange. If you are unable to control it via SmartThings, you must remove it from your network and add it again.",
+					eventType: "ALERT",
+					name: "secureInclusion",
+					value: "failed",
+					displayed: true,
+			)
+		}
+	} else {
+		def cmd = zwave.parse(description, [0x98: 1, 0x20: 1, 0x70: 1])
+		if (cmd) {
+			result = zwaveEvent(cmd)
+		}
+	}
+	log.debug "Parse returned ${result?.inspect()}"
+	return result
+}
+
+def zwaveEvent(securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand([0x20: 1, 0x85: 2, 0x70: 1])
+	// log.debug "encapsulated: $encapsulatedCommand"
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand)
+	}
+}
+
+def zwaveEvent(basicv1.BasicReport cmd) {
+	log.debug "rx $cmd"
+	handleDeviceEvent(cmd.value)
+}
+
+def zwaveEvent(sensorbinaryv2.SensorBinaryReport cmd) {
+	log.debug "rx $cmd"
+	def result = []
+
+	if (cmd.sensorType == sensorbinaryv2.SensorBinaryReport.SENSOR_TYPE_TAMPER) {
+		result << createEvent(name: "tamper", value: "detected")
+	} else {
+		result = handleDeviceEvent(cmd.sensorValue)
+	}
+	result
+}
+
+def zwaveEvent(notificationv3.NotificationReport cmd) {
+	def result = []
+
+	log.debug "rx $cmd"
+
+	if (cmd.notificationType == notificationv3.NotificationReport.NOTIFICATION_TYPE_BURGLAR) {
+		if (cmd.event == 3) {
+			result << createEvent(name: "tamper", value: "detected")
+		}
+	} else {
+		log.warn "Unknown cmd.notificationType: ${cmd.notificationType}"
+		result << createEvent(descriptionText: cmd.toString(), isStateChange: false)
+	}
+	result
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	log.debug "Unhandled Z-Wave command $cmd"
+	createEvent(displayed: false, descriptionText: "$device.displayName: $cmd")
+}
+
+def handleDeviceEvent(value) {
+	log.debug "handleDeviceEvent $value"
+	def result = [
+		createEvent([name: "switch", value: value ? "on" : "off", displayed: false]),
+		createEvent([name: "alarm", value: value ? "both" : "off"])
+	]
+
+	// value != 0 doesn't necessarily trigger the below events,
+	// but value == 0 clears them
+	if (!value) {
+		result << createEvent([name: "chime", value: "off"])
+		result << createEvent([name: "tamper", value: "clear"])
+	}
+
+	result
+}
+
+def generateCommand(command) {
+	def sound = (command && soundMap.containsKey(command)) ? soundMap[command] : soundMap[defaultSound]
+	log.debug "Sending $command"
+
+	secure(zwave.notificationV3.notificationReport(notificationType: sound.notificationType, event: sound.event))
+}
+
+def chime() {
+	log.debug "chime!"
+
+	// Chime is kind of special as the alarm treats it as momentary
+	// and thus sends no updates to us, so we'll send this and request an update
+	sendEvent(name: "chime", value: "chime")
+
+	[
+		generateCommand("Chime"),
+		secure(zwave.basicV1.basicGet())
+	]
+}
+
+def on() {
+	log.debug "sending on"
+	generateCommand(state.sound)
```

Comment:
Add a basic get + delay

Reply:
So I know I had this before from the Aeon code. A different question that @greens had asked had me also look into the necessity of the BasicGet. The device sense us a SensorBinaryReport in response to whatever command we send it. Should I add the BasicGet on top of that?

Reply:
Oh shoot, I did not see his comment. if that is the case, I would not. I actually dug around community and found https://github.com/krlaframboise/SmartThings/blob/master/devicetypes/krlaframboise/zipato-multisound-siren.src/zipato-multisound-siren.groovy 

and they were doing basic get in their DTH, and I was not 100% if the device does respond to commands.

Reply:
I don't know if that is the same hardware? (Though, side note: The Zipato device was what caused the Chime capability to be created.)

Reply:
It almost has to be the same hardware given that it is literally the same configuration/behavior.

Reply:
Ah, yeah, it is more complex of a device! 😮 But yeah, this device does report back with unsolicited responses. I'm not sure if the Zipato does as well. I suspect a lot of code is just copy&paste followed by "does it work? yup! ok!" At first I had somehow managed to miss the SensorBinaryReport messages.

Reply:
Ok, I just looked. It is the same device.

I see what they are doing. Since the device itself supports setting different sounds using NotificationReport instead of a configuration option (thus one packet to set the sound AND turn on the alarm), they are using strobe(), siren(), and both() independently. Since no one gave me any sort of "requirements" for this device and technically it ONLY supports both() (it blinks and makes noise regardless of what noise you ask it to make), I stuck with making all of the siren commands behave the same which is consistent with what we do for the Aeon Siren.